flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperator.java

```diff
@@ -652,11 +652,11 @@ private void processStreamStatus(StreamStatus streamStatus, int index) throws Ex
         }
     }
 
-    public final void processStreamStatus1(StreamStatus streamStatus) throws Exception {
+    public void processStreamStatus1(StreamStatus streamStatus) throws Exception {
         processStreamStatus(streamStatus, 0);
     }
 
-    public final void processStreamStatus2(StreamStatus streamStatus) throws Exception {
+    public void processStreamStatus2(StreamStatus streamStatus) throws Exception {
```

Comment:
Are these changes still needed?

Reply:
You mean the entire commit? I think it makes sense, otherwise the operator will generate `StreamStatus` for every record.

Reply:
No I meant the removal of `final` in `processStreamStatus1/2`. Doesn't seem to be necessary or the fix is incomplete.

Reply:
I override these two methods in `WatermarkAssignerOperator` which extends from `AbstractStreamOperator`. Therefore I need to remove the `final` here.

Reply:
Scratch that. I need to remove `final` only from `processStreamStatus`.

Reply:
That's what I meant :p